slackbot.py

```diff
@@ -0,0 +1,61 @@
+import os
+import time
+import re
+import requests
+from slackclient import SlackClient
+from config import slack_bot_token 
+
+slack_client = SlackClient(slack_bot_token)
+mapbot_id = None
+
+RTM_READ_DELAY = 1
+EXAMPLE_COMMAND = "help"
+MENTION_REGEX = "^<@(|[pipWU].+?)>(.*)"
+
+# API endpoint currently running on localhost
+URL = "http://127.0.0.1:5000/chatbot/"
+
+def parse_bot_commands(slack_events):
+    for event in slack_events:
+        if event["type"] == "message" and not "subtype" in event:
+            user_id, message = parse_direct_mention(event["text"])
+            if user_id == mapbot_id:
+                return message, event["channel"]
+    return None, None
+
+def parse_direct_mention(message_text):
+    matches = re.search(MENTION_REGEX, message_text)
+    # the first group contains the username, the second group contains the remaining message
+    return (matches.group(1), matches.group(2).strip()) if matches else (None, None)
+
+def handle_command(command, channel):
+    """
+        Executes bot command if command is known
+    """
+    default_response = "Not sure what you mean. Try *{}*.".format(EXAMPLE_COMMAND)
+
+    response = None
+    if command.startswith(EXAMPLE_COMMAND):
+        response = "Hey There! I am MapBot. Nice to meet you"
+    
+    else:
+        response = requests.get(url= URL+str(command)).json()
```

Comment:
So, what I see here is that this is working as an independent service. This would fail to run in case our bot itself is down. Do we want this separation? 
Assuming that we do, I would imagine the slack bot (and other bots) spinning up along with our original service and staying in some sort of sync (while maintaining independence) while running. 
@vishakha-lall would this require a separate container and orchestration between multiple containers (do we need to take a look at Kubernetes)? Or do we handle it in a fairly simple way where each bot integration returns a message in case it is not able to connect to the core service?

Reply:
@janakrajchadha Kubernetes is a usually a production level orchestrator, not recommended for development. This use case can be handled with `docker-compose`. 
I like the idea of the integrations starting up with the app, which for now could be linked to the service. Please do the needful @Rits1272